### PR TITLE
Added XCA (https://www.hohnstaedt.de/xca/)

### PR DESCRIPTION
**Pull request by edgardmessias**

### CODE REVIEW

**Comment by Ash258** [comment]
Why don't use portable version??
- Add hash extaction from <https://www.hohnstaedt.de/xca/index.php/download>

**Comment by edgardmessias** [comment]
I don't use the portable version to allow install old version in my machine, like 1.4.1 for old projects.

About hashs, the page uses `javascript` and `ajax`, I will try extract from here: https://raw.githubusercontent.com/chris2511/xca/master/misc/binary-hashes.json

File: xca.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+            "xca.exe",
+            "XCA - Certificate and Key management"
+        ]
+    ],
+    "checkver": {
+        "github": "https://github.com/chris2511/xca",
+        "regex": "/releases/tag/RELEASE\\.([\\d.]+)"
+    },
+    "autoupdate": {
+        "url": "https://github.com/chris2511/xca/releases/download/RELEASE.$version/setup_xca-$version.exe#dl.7z"
```
Comment (Ash258) [inline xca.json]:
```suggestion
        "url": "https://github.com/chris2511/xca/releases/download/RELEASE.$version/setup_xca-$version.exe#dl.7z",
        "hash": {
            "url": "https://raw.githubusercontent.com/chris2511/xca/master/misc/binary-hashes.json",
            "jsonpath": "$.['$basename']"
        }
```

Comment (edgardmessias) [reply xca.json]: Done

**Comment by Ash258** [comment]
> I don't use the portable version to allow install old version in my machine, like 1.4.1 for old projects.

I don't understand. If you extract exe you will get completely same files as in portable zip file. (Only installer dlls are not included)

Compare

**Comment by edgardmessias** [comment]
Update to use portable version or update to remove installer dlls?

**Comment by Ash258** [comment]
There is no reason to extract exe as 7zip, when there is zip provided with hashes by vendor.

**Comment by edgardmessias** [comment]
Updated to use ZIP

File: xca.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
+{
+    "homepage": "https://www.hohnstaedt.de/xca/",
+    "description": "X - Certificate and Key management",
+    "license": "BSD-3-Clause",
+    "version": "2.1.2",
+    "url": "https://github.com/chris2511/xca/releases/download/RELEASE.2.1.2/xca-portable-2.1.2.zip",
+    "hash": "37d2e315dc36c7060a50dd7203339dfa8c8b43cea34f340d456a900a6e5232a6",
+    "post_install": "Remove-Item \"$dir\\`$PLUGINSDIR\" -Recurse",
```
Comment (Ash258) [inline xca.json]: Remove redundant post_install after change to zip

```suggestion
    "extract_dir": "xca-portable-2.1.2",
```

Comment (r15ch13) [inline xca.json]: This can be removed

File: xca.json [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    "checkver": {
+        "github": "https://github.com/chris2511/xca",
+        "regex": "/releases/tag/RELEASE\\.([\\d.]+)"
+    },
+    "autoupdate": {
+        "url": "https://github.com/chris2511/xca/releases/download/RELEASE.$version/xca-portable-$version.zip",
+        "hash": {
+            "url": "https://raw.githubusercontent.com/chris2511/xca/master/misc/binary-hashes.json",
+            "jsonpath": "$.['$basename']"
+        }
```
Comment (Ash258) [inline xca.json]:
```suggestion
        },
        "extract_dir": "xca-portable-$version"
```

**Comment by edgardmessias** [comment]
Fixed extract_dir and remover redundant code

**Pull request merged**